### ISSUE TITLE
etcd3 support

### ISSUE THREAD

**Issue by danehans** [coreos/flannel#666]
In reviewing the flannel documentation, I am unable to determine of etcd3 is definitively supported. Can someone confirm if etcd3 is supported? Any guidance on this issue is appreciated.

**Comment by tomdee** [comment]
You can use the latest version of etcd with flannel but flannel still stores the data in the v2 format.

**Comment by danehans** [comment]
Thanks for the information. I am using the v3.1.0 etcd3 release. I'll try v3.1.5. I am using flannel v0.6.2. Let me know if I need to upgrade flannel as well.

I am trying to use the [matchbox examples](https://github.com/coreos/matchbox/tree/master/examples) to back a Kubernetes/Flannel cluster with etcd3.

cc/ @dghubble

**Comment by danehans** [comment]
I found my issue to be flanneld's interface discovery mechanism:

```
flannel-wrapper[2672]: I0411 20:01:55.284739 02672 manager.go:163] Using x.x.x.x as external interface
flannel-wrapper[2672]: I0411 20:01:55.284766 02672 manager.go:164] Using x.x.x.x as external endpoint
```

My k8s nodes have multiple interfaces. Flannel binds to the interface associated to the host's default route. In my instance, a node's public interface has the default route and was being used by flanneld instead of the private interface used for node<>node communication (i.e. pod networking). 

I fixed the issue by specifying the `--FLANNELD_IFACE=` private interface in flanenl.env.

Thanks for your help.
